hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/BalancingPolicy.java

```diff
@@ -138,7 +138,7 @@ String getName() {
     void accumulateSpaces(DatanodeStorageReport r) {
       for(StorageReport s : r.getStorageReports()) {
         final StorageType t = s.getStorage().getStorageType();
-        totalCapacities.add(t, s.getCapacity());
+        totalCapacities.add(t, s.getRemaining() + s.getBlockPoolUsed());
```

Comment:
IMO, it's difficult to understand the meaning of this improvement, just suggest add some annotations for Pool policy and describe what and why.